lib/webauthn/attestation.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require "cbor"
+require "openssl"
+require "webauthn/attestation_statement"
+require "webauthn/authenticator_data"
+
+module WebAuthn
+  class Attestation
+    def self.deserialize(attestation_object)
+      from_map(CBOR.decode(attestation_object))
+    end
+
+    def self.from_map(map)
+      new(
+        authenticator_data: WebAuthn::AuthenticatorData.new(map["authData"]),
+        attestation_statement: WebAuthn::AttestationStatement.from(map["fmt"], map["attStmt"])
```

Comment:
Ideally this would also be a call to a `#deserialize` method instead of `.new`.

Reply:
> Ideally this would also be a call to a `.deserialize` method instead of `.from`.

Scratch this.

`.deserialize` doesn't make sense because `map["attStmt"]` is a "map", not a byte string, as the case of `map["authData"]`.

So ideally this would be named `.from_map`, not `.deserialize`.